sql/core/src/main/scala/org/apache/spark/sql/execution/datasources/jdbc/JdbcUtils.scala

```diff
@@ -328,21 +328,25 @@ object JdbcUtils extends Logging with SQLConfHelper {
   /**
    * Convert a [[ResultSet]] into an iterator of Catalyst Rows.
    */
-  def resultSetToRows(resultSet: ResultSet, schema: StructType): Iterator[Row] = {
+  def resultSetToRows(
+      resultSet: ResultSet,
+      schema: StructType,
+      dialect: Option[JdbcDialect] = None): Iterator[Row] = {
```

Comment:
Why is dialect optional? I think you can just pass dialect as JdbcDialect.

Reply:
I am not sure how and where this function is used.
As it is a public function, I am thinking maybe we want to keep backward compatibilities?

Reply:
For backward compatibility, this solution would require a separate constructor, just having the default value would not work in Java.